controls/SV-204445.rb

```diff
@@ -1,57 +1,56 @@
-    # more /etc/cron.daily/aide
-    #!/bin/bash
-    /usr/sbin/aide --check | /bin/mail -s "$HOSTNAME - Daily aide integrity check run" root@sysname.mil'
+
+Note: A file integrity tool other than Advanced Intrusion Detection Environment (AIDE) may be used, but the tool must be executed at least once per week.
+
+Check for the presence of a cron job running daily or weekly on the system that executes AIDE daily to scan for changes to the system baseline. The command used in the example will use a daily occurrence.
+
+Check the cron directories for a script file controlling the execution of the file integrity application. For example, if AIDE is installed on the system, use the following command:
+
+     # ls -al /etc/cron.* | grep aide
+     -rwxr-xr-x 1 root root 602 Mar 6 20:02 aide
+
+     # grep aide /etc/crontab /var/spool/cron/root
+     /etc/crontab: 30 04 * * * root /usr/sbin/aide  --check
+     /var/spool/cron/root: 30 04 * * * /usr/sbin/aide  --check
+
+If the file integrity application does not exist, or a script file controlling the execution of the file integrity application does not exist, this is a finding.'
+  desc 'fix', 'Configure the file integrity tool to run automatically on the system at least weekly. The following example output is generic. It will set cron to run AIDE daily, but other file integrity tools may be used:
+
+     # more /etc/cron.daily/aide
+     #!/bin/bash
+
+     /usr/sbin/aide --check | /var/spool/mail -s "$HOSTNAME - Daily aide integrity check run" root@sysname.mil'
   impact 0.5
-  tag 'legacy': ['SV-86597', 'V-71973']
-  tag 'severity': 'medium'
-  tag 'gtitle': 'SRG-OS-000363-GPOS-00150'
-  tag 'gid': 'V-204445'
-  tag 'rid': 'SV-204445r603261_rule'
-  tag 'stig_id': 'RHEL-07-020030'
-  tag 'fix_id': 'F-36304r602622_fix'
-  tag 'cci': ['CCI-001744']
+  tag legacy: ['SV-86597', 'V-71973']
+  tag severity: 'medium'
+  tag gtitle: 'SRG-OS-000363-GPOS-00150'
+  tag gid: 'V-204445'
+  tag rid: 'SV-204445r880848_rule'
+  tag stig_id: 'RHEL-07-020030'
+  tag fix_id: 'F-36304r880847_fix'
+  tag cci: ['CCI-001744']
   tag nist: ['CM-3 (5)']
-  tag subsystems: ["file_integrity_tool"]
-  tag 'host', 'container'
+  tag subsystems: ['file_integrity_tool']
+  tag 'host'
+  tag 'container'
 
   file_integrity_tool = input('file_integrity_tool')
   file_integrity_interval = input('file_integrity_interval')
 
-  describe package(file_integrity_tool) do
-    it { should be_installed }
-  end
-
-  if file_integrity_interval == 'monthly'
+  if file_integrity_tool != 'aide'
+    describe 'The system is not set to ' do
+      skip "The system is set to a `disconnected` state and you must validate
+        the state of the system packages manually, or through another process, if you
+        have an established update and patch process, please set this control as
+        `Not Applicable` with a `caevat` via an overlay."
+    end
+  elsif file_integrity_interval == 'monthly'
```

Comment:
"Chef InSpec will always evaluate all the tests contained within describe.one. It does not short-circuit upon evaluating a passing describe block."  however the tests displayed only have the one result listed in the case of the pass.  whereas in the case of the fail it lists out each fail.  if each describe is evaluated in the describe.one, shouldn't the correct behavior for the overarching pass be to mark any passing describe blocks as passed and any faileds as skips?  this is a question based outta my inspec ignorance and a feeling of uneasiness caused by the number of subtests being different for no apparent reason to me the end user.  like i had manually corrected only the one case there and it magically fixed the entire control i'd be pretty surprised lol.

<img width="1722" alt="image" src="https://github.com/mitre/redhat-enterprise-linux-7-stig-baseline/assets/2015821/fa081217-aee5-42ee-945c-2d614f47cb4d">

Reply:
Yes, I saw this behavior in another control I was testing earlier with describe.one. For example, there was a control saying you must have either screen or tmux installed. If it fails, it states no screen installed and no tmux installed. If you install screen, it just says you pass because screen is installed and does not give you a failure that tmux is not installed because it only needs one of the checks to be true. Does that help this comment?

Reply:
What happens if you had both screen and tmux installed?  Does it only give the subtest result for the first one it attempted to assess or does it give the "extra credit" of the double pass?

Reply:
It gives one pass.